dotnet/src/Functions/Functions.Yaml/PromptExecutionSettingsTypeConverter.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json.Serialization;
+using YamlDotNet.Core;
+using YamlDotNet.Core.Events;
+using YamlDotNet.Serialization;
+using YamlDotNet.Serialization.BufferedDeserialization;
+using YamlDotNet.Serialization.NamingConventions;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Allows custom deserialization for <see cref="PromptExecutionSettings"/> from YAML prompts.
+/// </summary>
+internal sealed class PromptExecutionSettingsTypeConverter : IYamlTypeConverter
+{
+    /// <inheritdoc/>
+    public bool Accepts(Type type)
+    {
+        return type == typeof(PromptExecutionSettings);
+    }
+
+    /// <inheritdoc/>
+    public object? ReadYaml(IParser parser, Type type)
```

Comment:
Wonder if they have `async` methods for parsing, and using those instead on synchronous ones.

```suggestion
    public object? ReadYaml(IParser parser, Type type)
```

Reply:
Nope, unfortunately the `IYamlTypeConverter` interface does not have async methods:
<img width="777" alt="image" src="https://github.com/user-attachments/assets/67e815da-00c5-450a-881b-190a3237200f">